### ISSUE TITLE
results not matching other hr deficiency tools

### ISSUE THREAD

**Issue by anoronh4** [UMCUGenetics/CHORD#10]
I have heard great things about CHORD and I'm really excited to try it. So far the installation works and I can run it, but the results do not match my expectations. For example, I am attaching the results of two samples, one is BRCA wildtype and the other is BRCA mutated. These samples have been identified as HR deficient/proficient in a way that is expected by HRDetect, but CHORD calls them both as proficient, even with the same SNP/Indel and SV inputs. 
[BRCA-mt.chord.txt](https://github.com/UMCUGenetics/CHORD/files/9154377/BRCA-mt.chord.txt)
[BRCA-wt.chord.txt](https://github.com/UMCUGenetics/CHORD/files/9154378/BRCA-wt.chord.txt)

Just wondering if there are any obvious issues in the attached files, or if there's any debugging or additional modifications that we can do from here to get better results.

**Comment by luannnguyen** [comment]
Hi,

Thanks for your interest in CHORD. The files seem to be ok.

The mutant sample seems to be an edge case (from the perspective of CHORD). This sample only had ~6% del.mh.bimh.2.5 (deletions with >=2bp flanking homology). Typically CHORD considers samples as HRD if they have >15-20% del.mh.bimh.2.5. This sample does have 15% del.mh.bimh.1 (deletions with 1bp flanking homology), which is moderately high but not super high, so there is some indication that this sample could be HRD. However, del.mh.bimh.1 has less weight than del.mh.bimh.2.5 for predicting HRD, and is also associated with radiotherapy.

```
## BRCA mutated sample
del.rep	0.093577982
ins.rep	0.016513761
del.mh.bimh.1	0.15412844
del.mh.bimh.2.5	0.057798165
ins.mh	0.017431193
del.none	0.585321101
ins.none	0.075229358
```

HRDetect likely predicts the mutant sample as HRD because the model does not split microhomology deletions by length of flanking homology.

Does this mutant sample have a single mutation in BRCA, or biallelic loss of BRCA? If the former, it could be that this sample is not actually HRD.

Luan

**Comment by anoronh4** [comment]
Yes, this sample has a biallelic loss of function in BRCA2, and i chose it because it had the highest HRDetect score among 24 BRCA-deficient samples. In fact, all 24 of my BRCA-deficient samples were determined as `HR_proficient` by CHORD. HRDetect says `del.mh.prop` is .34 -- not sure how familiar you are with that tool, but should we expect `del.mh.bimh.2.5` + `del.mh.bimh.1` to approximate the HRDetect number? 

i am now realizing what the issue might be. I am using the our file as input and extracting relevant columns, but the maf represents the ref and alt differently from a vcf file. For example in the maf a variant would show as:
```
CHROM	POS	REF	ALT
1	726207	GAATGGAATGGACTC	-
```
Whereas extracting it from the vcf it would come out like this:
```
CHROM	POS	REF	ALT
1	726207	GGAATGGAATGGACTC	G
```
After reformatting the ref/alt positions, I was able to determine HR_deficiency in the sample:
```
cat BRCA-mt.chord.txt | cut -f 1-5,17-18
sample	p_BRCA1	p_BRCA2	p_hrd	hr_status	del.mh.bimh.1	del.mh.bimh.2.5
sample	0.15	0.638	0.788	HR_deficient	0.074235807860262	0.369931378665003
```
so i think that's the solution, although curiously all HR deficient samples are BRCA2 type.

**Comment by luannnguyen** [comment]
Ok, so if i understand correctly the homology length in your old files would then always 1 less than it should be. You would then have fewer del mh in total, and this would indeed explain why the samples were not being predicted as HRD :)

I've never tested whether del.mh.bimh.2.5 + del.mh.bimh.1 is approx equal to del.mh.prop. I would assume that they are linearly correlated, so CHORD del mh would be within the same order of magnitude as HRDetect del mh.

**Comment by anoronh4** [comment]
I'm not sure if that works out perfectly that way, since you would expect the amount subtracted from the old `del.mh.bimh.1` would be added to the new `del.mh.bimh.2.5`, right? `del.mh.bimh.1`(old) - `del.mh.bimh.1`(new) = 0.154 - 0.074 = .08 , and `del.mh.bimh.2.5`(old) - `del.mh.bimh.2.5`(new) = 0.058 - 0.370 = -0.312. Maybe I don't fully understand the logic here.

As a sidenote, i was unable to get a custom BSgenome working. When I used `BSgenome.Hsapiens.1000genomes.hs37d5` instead of the default, i got the following error: 
```
## Single base substitutions
Loading variants...
Initializing SNV signature output vector...
Removing rows with multiple ALT sequences...
Subsetting for SNVs...
Getting SNV trinucleotide contexts...
Error in .getOneSeqFromBSgenomeMultipleSequences(x, names[i], start[i],  : 
  sequence chr1 not found
Calls: extractSigsChord ... lapply -> FUN -> .getOneSeqFromBSgenomeMultipleSequences
In addition: Warning message:
In subsetSmnvs(df, type = "snv", verbose = verbose) :
  colnames(df)[1:4] != c('chrom','pos','ref','alt'). Assuming first 4 columns are these columns
Execution halted
```
I'm not sure how `chr1` would be an input error since my input files are not UCSC style (`1` rather than `chr1`), and the alternative build I am using also does not match that. I wonder if CHORD is trying to coerce my chromosome names when it's not necessary.

**Comment by luannnguyen** [comment]
Regarding your first point, my initial logic was indeed a bit oversimplified, but i don't think you can make such calculations on the relative counts. I think the increase in the number of length 1 indels resulted in more _absolute_ counts of all indel types, but in the end there was _proportionally/relatively_ more  `del.mh.bimh.2.5` (e.g. due to having relatively less `del.rep` and `ins.rep` of repeat length 1). 

Regarding the ref genome, `mutSigExtractor::variantsFromVcf()` indeed tries to force the chrom names to UCSC style. If you would manually force the ref genome chrom names to UCSC style, does the error go away?

**Comment by anoronh4** [comment]
Sure I can make the reference genome have "chr", or i can just use the UCSC one, so i will probably just end up doing that if all chromosomes match.
